pynio/block.py

```diff
@@ -37,6 +48,47 @@ def name(self):
     def type(self):
         return self._type
 
+    @property
+    def template(self):
+        return self._template
+
+    @template.setter
+    def template(self, value):
+        # do type checking later
+        self._template = value
+
+    @property
+    def config(self):
+        return self._config
+
+    @config.setter
+    def config(self, value):
+        if self._template is None:
+            self._config = value
+            return
+        config = deepcopy(self._template)
+        config.readonly = False
+        config.update(value, drop_unknown=True,
+                      drop_logger=self._instance.droplog)
+        self._config = config
+
+    def json(self):
+        return self._config.__basic__()
+
+    def _load_template(self, type, value, instance=None):
```

Comment:
Why does type have to be passed to load_template? Doesn't the Block already know it's type?